apps/server/src/serializers/entity.ts

```diff
@@ -39,16 +50,39 @@ export const EntitySerializer = serializer({
         )
       : {};
 
+    const parentIds = itemList.map((i) => i.parentId).filter(notEmpty);
+    const parentList = parentIds.length
+      ? await db.select().from(entities).where(inArray(entities.id, parentIds))
+      : [];
+
+    // Use a simplified approach to avoid circular reference
+    const parentsById: Record<number, { id: number; name: string }> = {};
+
+    for (const parent of parentList) {
+      parentsById[parent.id] = {
+        id: parent.id,
+        name: parent.name,
+      };
+    }
+
     return Object.fromEntries(
-      itemList.map((item) => {
-        return [
-          item.id,
-          {
-            country: item.countryId ? countriesById[item.countryId] : null,
-            region: item.regionId ? regionsById[item.regionId] : null,
-          },
-        ];
-      }),
+      itemList.map(
+        (item): [number, { country: any; region: any; parent: any | null }] => {
+          return [
+            item.id,
+            {
+              country: item.countryId ? countriesById[item.countryId] : null,
+              region: item.regionId ? regionsById[item.regionId] : null,
+              parent: item.parentId
```

Comment:
this should be a serializer result of EntitySerializer, similar to how we do it elsewhere